app/views/concerns/searchable_view.rb

```diff
@@ -43,42 +39,49 @@ def input_search
   def input_search_values(*keys)
     return [] if keys.blank?
 
-    keys.map do |k|
-      [params[k]].flatten.compact.reject(&:blank?).map do |v|
-        remove_params = remove_search_param(k, v, params)
-        remove_params[:q] = '' if (k.to_s == 'mlt') && !remove_params.key?(:q)
-        {
-          name: params[k].is_a?(Array) ? "#{k}[]" : k.to_s,
-          is_mlt: k.to_s == 'mlt',
-          value: k.to_s == 'mlt' ? params[:mlt] : input_search_param_value(k, v),
-          remove: search_action_path(remove_params)
-        }
-      end
+    keys.map do |param_key|
+      input_search_value_one_param(param_key)
     end.flatten.compact
   end
 
-  ##
-  # Returns text to display on-screen for an active search param
-  #
-  # @param key [Symbol] parameter key
-  # @param value value of the parameter
-  # @return [String] text to display
-  def input_search_param_value(key, value)
-    case key
-    when :mlt
-      response, doc = controller.fetch(value)
-      item = render_index_field_value(doc, ['dcTitleLangAware', 'title'])
-      t('site.search.similar.prefix', mlt_item: item)
+  def input_search_value_one_param(local_key, local_params = params, params_key = local_key)
```

Comment:
I think it might be clearer to rename this `input_search_value_from_param(_key)` instead of `input_search_value_one_param` 
If this is renamed `input_search_value_one_param_field` should also be renamed to `input_search_value_from_param_field` too.